core/src/main/java/org/infinispan/persistence/sifs/NonBlockingSoftIndexFileStore.java

```diff
@@ -510,7 +516,10 @@ protected boolean isIndexLoaded() {
    @Override
    public CompletionStage<Long> approximateSize(IntSet segments) {
       // Approximation doesn't pause the appender so the index can be slightly out of sync
-      return CompletableFuture.completedFuture(index.approximateSize(segments));
+      long totalSize = index.approximateSize();
+      IntSet matchingSegments = IntSets.mutableCopyFrom(segments);
+      matchingSegments.retainAll(matchingSegments);
+      return CompletableFuture.completedFuture(totalSize * matchingSegments.size() / ownedSegments.size());
```

Comment:
I would say we can use the TemporaryTable to store how many segments it owns instead, but I am fine with this impl.

Reply:
Done